python/tests/expressions/test_expressions_base.py

```diff
@@ -300,138 +337,289 @@ def test_strs(op, string):
     assert str(op) == string
 
 
-def test_ref_binding_case_sensitive(request):
-    schema = request.getfixturevalue("table_schema_simple")
-    ref = base.Reference("foo")
-    bound = base.BoundReference(schema.find_field(1), schema.accessor_for_field(1))
-    assert ref.bind(schema, case_sensitive=True) == bound
+def test_ref_binding_case_sensitive(table_schema_simple: Schema):
+    ref = base.Reference[str]("foo")
+    bound = base.BoundReference[str](table_schema_simple.find_field(1), table_schema_simple.accessor_for_field(1))
+    assert ref.bind(table_schema_simple, case_sensitive=True) == bound
 
 
-def test_ref_binding_case_sensitive_failure(request):
-    schema = request.getfixturevalue("table_schema_simple")
-    ref = base.Reference("Foo")
+def test_ref_binding_case_sensitive_failure(table_schema_simple: Schema):
+    ref = base.Reference[str]("Foo")
     with pytest.raises(ValueError):
-        ref.bind(schema, case_sensitive=True)
+        ref.bind(table_schema_simple, case_sensitive=True)
 
 
-def test_ref_binding_case_insensitive(request):
-    schema = request.getfixturevalue("table_schema_simple")
-    ref = base.Reference("Foo")
-    bound = base.BoundReference(schema.find_field(1), schema.accessor_for_field(1))
-    assert ref.bind(schema, case_sensitive=False) == bound
+def test_ref_binding_case_insensitive(table_schema_simple: Schema):
+    ref = base.Reference[str]("Foo")
+    bound = base.BoundReference[str](table_schema_simple.find_field(1), table_schema_simple.accessor_for_field(1))
+    assert ref.bind(table_schema_simple, case_sensitive=False) == bound
 
 
-def test_ref_binding_case_insensitive_failure(request):
-    schema = request.getfixturevalue("table_schema_simple")
-    ref = base.Reference("Foot")
+def test_ref_binding_case_insensitive_failure(table_schema_simple: Schema):
+    ref = base.Reference[str]("Foot")
     with pytest.raises(ValueError):
-        ref.bind(schema, case_sensitive=False)
+        ref.bind(table_schema_simple, case_sensitive=False)
 
 
 def test_in_to_eq():
     assert base.In(base.Reference("x"), (literal(34.56),)) == base.EqualTo(base.Reference("x"), literal(34.56))
 
 
-def test_bind_in(request):
-    schema = request.getfixturevalue("table_schema_simple")
+def test_empty_bind_in(table_schema_simple: Schema):
+    bound = base.BoundIn[str](
```

Comment:
I found this test case slightly awkward since you create a `BoundIn`, but because it is empty, it gets changed into an `AlwaysFalse`

Reply:
Yeah, it's a bit odd.